File: adventofcode-2024/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day11/Part1.java
package nl.mvdr.adventofcode.adventofcode2024.day11;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Part1 extends PlutonianPebblesSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    public Part1() {
        super(25);
    }
    
    public static void main(String[] args) {
        var instance = new Part1();

        var result = instance.solve("input-day11-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day09/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day09

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.point.Point
import nl.mvdr.adventofcode.solver.FunctionSolver
import kotlin.math.abs

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>): Long {
    val points = lines.map(Point::parse).toSet()

    return points.flatMap { firstCorner -> points.map { secondCorner -> Rectangle(firstCorner, secondCorner) } }
        .maxOf(Rectangle::area)
}

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day09.txt")
    logger.info { result }
}
